Tella/Domain/Entity/Report/Vault/VaultFileDB.swift

```diff
@@ -0,0 +1,293 @@
+        self.mimeType = vaultFile.fileExtension.mimeType()
+        self.created = vaultFile.created
+        if let width = vaultFile.resolution?.width {
+            self.width = width
+        }
+        if let height = vaultFile.resolution?.height {
+            self.height = height
+        }
+    }
+    
+    required init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        
+        id = try container.decode(String.self, forKey: .id)
+        
+        let typeInt = try container.decode(Int.self, forKey: .type)
+        type = VaultFileType(rawValue: typeInt) ?? .unknown
+        
+        thumbnail = try container.decode(Data.self, forKey: .thumbnail)
+        name = try container.decode(String.self, forKey: .name)
+        
+        let createdDouble = try container.decode(Double.self, forKey: .created)
+        created = createdDouble.getDate() ?? Date()
+        
+        duration = try container.decode(Double.self, forKey: .duration)
+        size = try container.decode(Int.self, forKey: .size)
+        
+        mimeType = try container.decode(String.self, forKey: .mimeType)
+        
+        width = try container.decode(Double.self, forKey: .width)
+        height = try container.decode(Double.self, forKey: .height)
+        
+        
+    }
+    
+    init(dictionnary: [String:Any])   {
+        
+        let id = dictionnary[CodingKeys.id.rawValue] as? String
+        
+        let typeInt = dictionnary[CodingKeys.type.rawValue] as? Int
+        let type = VaultFileType(rawValue: typeInt ?? 0) ?? .unknown
+        
+        
+        let thumbnail = dictionnary[CodingKeys.thumbnail.rawValue] as? Data
+        let name = dictionnary[CodingKeys.name.rawValue] as? String
+        
+        
+        let createdDouble = dictionnary[CodingKeys.created.rawValue] as? Double
+        let created = createdDouble?.getDate() ?? Date()
+        
+        
+        let duration = dictionnary[CodingKeys.duration.rawValue] as? Double
+        
+        
+        let size = dictionnary[CodingKeys.size.rawValue] as? Int
+        let mimeType = dictionnary[CodingKeys.mimeType.rawValue] as? String
+        
+        let width = dictionnary[CodingKeys.width.rawValue] as? Double
+        let height = dictionnary[CodingKeys.height.rawValue] as? Double
+        
+        
+        
+        
+        self.id = id ?? ""
+        self.type = type
+        self.thumbnail = thumbnail
+        self.name = name ?? ""
+        self.created = created
+        self.duration = duration
+        self.size = size ?? 0
+        self.mimeType = mimeType
+        self.width = width
+        self.height = height
+        
+        
+    }
+}
+
+extension VaultFileDB: CustomDebugStringConvertible {
+    
+    var debugDescription: String {
+        return "\(type): \(String(describing: name)), \(id)"
+    }
+    
+}
+
+
+extension VaultFileDB {
+    
+    var thumbnailImage: UIImage {
```

Comment:
I don't think that we can do a cache on this because it's just an initialisation for an UIImage with data and already we have the data in the database, but if you have a suggestion you can share it.